sasgrid-and-viya/azuredeploy.json

```diff
@@ -0,0 +1,5833 @@
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/networkSecurityGroups', variables('oss_nw_sg'))]",
+                "[resourceId('Microsoft.Network/virtualNetworks',variables('vnet_name'))]"
+            ],
+            "properties": {
+                "ipConfigurations": [
+                    {
+                        "name": "ipconfig1",
+                        "properties": {
+                            "privateIPAllocationMethod": "Dynamic",
+                            "subnet": {
+                                "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('vnet_name'), variables('vnet_lustre_pvt_subnt'))]"
+                            },
+                            "privateIPAddressVersion": "IPv4"
+                        }
+                    }
+                ],
+                "enableAcceleratedNetworking": true,
+                "enableIPForwarding": false,
+                "networkSecurityGroup": {
+                    "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('oss_nw_sg'))]"
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines",
+            "apiVersion": "2019-07-01",
+            "name": "[concat(variables('oss_vm_name'), copyIndex(1))]",
+            "location": "[parameters('location')]",
+            "identity": {
+                "type": "SystemAssigned"
+            },
+            "copy": {
+                "name": "sasnodecopy",
+                "count": "[parameters('numberOfLustreOSSNodes')]"
+            },
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/networkInterfaces', concat(parameters('SASApplicationName'),'_',variables('oss_vm_name'),copyIndex(1),'_nic'))]",
+                "[resourceId('Microsoft.Storage/storageAccounts', variables('diagnostic_storagegroup_name'))]"
+            ],
+            "tags": "[variables('lustrevm_tags')]",
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('lustreOSSNodeVmSize')]"
+                },
+                "storageProfile": {
+                    "imageReference": {
+                        "publisher": "RedHat",
+                        "offer": "RHEL",
+                        "sku": "7.7",
+                        "version": "latest"
+                    },
+                    "osDisk": {
+                        "osType": "Linux",
+                        "name": "[concat(variables('oss_vm_name'),copyIndex(1),'-osdisk')]",
+                        "createOption": "FromImage",
+                        "caching": "ReadWrite",
+                        "managedDisk": {
+                            "storageAccountType": "Premium_LRS"
+                        },
+                        "diskSizeGB": "[variables('lustre_osdisk_size')]"
+                    },
+                    "dataDisks": [
+                        {
```

Comment:
use a copy loop for these

Reply:
We are provisioning only one data disk for  different virtual machines(mgt, mdt)  and hence didn't used copy loop.

Reply:
modified the code, used Copy loop.